UnityProject/Assets/Scripts/Gateway/StationGateway.cs

```diff
@@ -0,0 +1,215 @@
+			{
+				if (isOn == true)
+				{
+					SetOnline();
+				}
+				else if (isOn == false)
+				{
+					SetOffline();
+				}
+				timeElapsedClient = 0;
+			}
+		}
+	}
+
+	private void OnEnable()
+	{
+		UpdateManager.Add(CallbackType.UPDATE, UpdateMe);
+	}
+	void OnDisable()
+	{
+		UpdateManager.Remove(CallbackType.UPDATE, UpdateMe);
+	}
+
+	private void Start()
+	{
+		SetOffline();
+
+		if (!isServer) return;
+
+		registerTile = GetComponent<RegisterTile>();
+		Position = registerTile.WorldPosition;
+
+		ServerChangeState(false);
+		var count = Random.Range(RandomCountBegining, RandomCountEnd);
+		Invoke(nameof(WorldSetup), count);
+	}
+
+	[Server]
+	private void WorldSetup()
+	{
+		//Selects Random world
+		SelectedWorld = Worlds[Random.Range(0, Worlds.Count)];
+
+		if (SelectedWorld == null) return;
+
+		var selectedWorld = SelectedWorld.GetComponent<WorldGateway>();
+
+		Message = "Teleporting to: " + selectedWorld.WorldName;
+
+		if (selectedWorld.IsOnlineAtStart == false)
+		{
+			selectedWorld.IsOnlineAtStart = true;
+			selectedWorld.SetUp();
+		}
+
+		if (HasPower == true)
+		{
+			SetOnline();
+			ServerChangeState(true);
+
+			var text = "Alert! New Gateway connection formed.\n\n Connection established to: " + SelectedWorld.GetComponent<WorldGateway>().WorldName;
+			CentComm.MakeAnnouncement(CentComm.CentCommAnnounceTemplate, text, CentComm.UpdateSound.alert);
+		}
+	}
+
+	[Server]
+	private void DetectPlayer()
+	{
+		//detect players positioned on the portal bit of the gateway
+		var playersFound = Matrix.Get<ObjectBehaviour>(registerTile.LocalPositionServer + Vector3Int.up, ObjectType.Player, true);
+
+		foreach (ObjectBehaviour player in playersFound)
+		{
+			var coord = new Vector2(Position.x, Position.y);
+			Chat.AddLocalMsgToChat(Message, coord, gameObject);
+			SoundManager.PlayNetworkedForPlayer(player.gameObject, "StealthOff"); //very weird, sometimes does the sound other times not.
+			TransportPlayers(player);
+		}
+
+		foreach (var objects in Matrix.Get<ObjectBehaviour>(registerTile.LocalPositionServer + Vector3Int.up, ObjectType.Object, true))
+		{
+			TransportObjects(objects);
+		}
+
+		foreach (var items in Matrix.Get<ObjectBehaviour>(registerTile.LocalPositionServer + Vector3Int.up, ObjectType.Item, true))
+		{
+			TransportItems(items);
+		}
+	}
+
+	[Server]
+	private void TransportPlayers(ObjectBehaviour player)
+	{
+		//teleports player to the front of the new gateway
+		player.GetComponent<PlayerSync>().SetPosition(SelectedWorld.GetComponent<RegisterTile>().WorldPosition, false);
+	}
+
+	[Server]
+	private void TransportObjects(ObjectBehaviour objects)
+	{
+		objects.GetComponent<CustomNetTransform>().SetPosition(SelectedWorld.GetComponent<RegisterTile>().WorldPosition);
+	}
+
+	[Server]
+	private void TransportItems(ObjectBehaviour items)
+	{
+		items.GetComponent<CustomNetTransform>().SetPosition(SelectedWorld.GetComponent<RegisterTile>().WorldPosition);
+	}
+
+	private void SetOnline()
+	{
```

Comment:
You can avoid those 3 methods by making one method in which you pass Sprite[]
private void SetOnline(Sprite[] sprites)

Reply:
Doesnt seem too necessary to change atm.